### ISSUE TITLE
How to train the models?

### ISSUE THREAD

**Issue by yikuanglancheng** [ZehaoDou-official/FPS-SMC-2023#1]
Hi! Thanks for your impressive work, but how should the model be trained? Can you provide the code and hyperparameter configurations for the training part?

**Comment by ZehaoDou-official** [comment]
Hello. FPS does not have any model training part, just like DPS. In diffusion models, the model training is mainly on the unconditional score matching, where we directly apply pretrained score models. We have clearly listed all the pretrained score models we use on FFHQ and ImageNet datasets in the README file. If you are interested in the training of score models, please refer to https://github.com/openai/guided-diffusion.

**Comment by yikuanglancheng** [comment]
Thank you for your answer. Can I understand that for a specific dataset, if I want to use FPS for deblurring or inpainting tasks, I need to pre-train a regular image generation task on the original dataset first? Is this pre-trained model unconditional or conditional?

**Comment by ZehaoDou-official** [comment]
Yes, the overall process has two stages: pre-training and sampling. First, we use unconditional score matching on the original dataset to train a standard image generation model. This pre-training step is always the same, regardless of the downstream task. Each dataset only needs one pre-trained unconditional score model, which can then be applied to various sampling-based tasks, such as inpainting or deblurring, without having to train a new score model for each task.

**Comment by yikuanglancheng** [comment]
Thank you!

**Comment by DrDLyuan** [comment]
In the sampling process of Proposition B.3, how is the operation (I+A^{T}A)^{-1} implemented in code? For example, if A is an inpainting operator, how would this operation be represented in code?
